addons/website_slides/models/slide_channel.py

```diff
@@ -453,6 +457,27 @@ def _get_default_enroll_msg(self):
     def _compute_enroll(self):
         self.filtered(lambda channel: channel.visibility == 'members').enroll = 'invite'
 
+    @api.depends('visibility', 'is_member')
+    @api.depends_context('uid')
+    def _compute_is_visible(self):
+        for channel in self:
+            if channel.visibility == 'public' or channel.is_member or (not self.env.user._is_public() and channel.visibility == 'connected'):
+                channel.is_visible = True
+            else:
+                channel.is_visible = False
+
+    @api.model
+    def _search_is_visible(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise NotImplementedError(_('Operation not supported'))
+        check_is_visible = value if operator == '=' else not value
+        if not self.env.user._is_public():
+            domain = ['|', ('is_member', '=', True), ('visibility', 'in', ['public', 'connected'])]
+        else:
+            domain = [('visibility', '=', 'public')]
+        channel_ids = self.env['slide.channel']._search(domain)
```

Comment:
Shouldn't public users only view the public visibility courses?

Reply:
Yes, I guess it's technicality the same, since public user can not be member, but to avoid any confusion I would still try to get the same exact logic for both
(so eg moving the member check in the compute, or changing the visibility check in the search)